### DISCUSSION TITLE
Migrate Freeform subscriptions to Craft Commerce

### DISCUSSION THREAD

**Discussion by redkrupp** [solspace/craft-freeform#126] [Questions & Answers] [answered]
Hey,

Will it work if I simply use already created, used and attached to freeform membership plan (stripe) as a new subscription plan in craft commerce. Do my existing users will be synced since their emails are already in the database? Any ideas or input points if nothing would happen? Thanks.

**Comment by gustavs-gutmanis** [comment, marked as answer]
I'm assuming it should work, but I cannot guarantee it will, we would have to test it.  
I guess the worst that could happen is that you don't get synced users...

**Comment by redkrupp** [comment]
Yup. I just tried, it hasn't synced users. Is there a workaround for this?

On Fri, Apr 2, 2021 at 1:31 PM Gustavs Gūtmanis ***@***.***>
wrote:

> I'm assuming it should work, but I cannot guarantee it will, we would have
> to test it.
> I guess the worst that could happen is that you don't get synced users...
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/solspace/craft3-freeform/discussions/126#discussioncomment-561376>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAL2YHAKTV4OJXWE2CFCNITTGVXGLANCNFSM42ILLTQQ>
> .
>


-- 
Regards,
Mark

**Reply by gustavs-gutmanis** [reply]
Currently Freeform doesn't sync users because we're never storing which user submitted the form, so we cannot know if they're the same person. And it's not always given that they will provide their email since that is up to the field mapping.

Looks like it isn't currently possible. You could write a module that handles your specific business logic by [handling several events](https://docs.solspace.com/craft/freeform/v3/developer/stripe-payments-events.html) that are fired by Freeform during the stripe integration process.

Hope that helps. Otherwise, you could create a feature request and we'll add it to the list 🙂
